wemake_python_styleguide/logic/arguments/function_args.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+
+import ast
+from itertools import zip_longest
+from typing import List, Optional
+
+from wemake_python_styleguide import types
+
+
+def get_args_without_self(node: types.AnyFunctionDefAndLambda) -> List[ast.arg]:
+    """Gets ``node`` arguments excluding ``self``."""
+    node_args = node.args.args
+    if not node_args:
+        return node_args
+    if isinstance(node, ast.Lambda):
+        return node_args
+    first_arg = node_args[0]
+    if first_arg.arg != 'self':
+        return node_args
+    return node_args[1:]
+
+
+def has_same_vararg(
+    node: types.AnyFunctionDefAndLambda,
+    call: ast.Call,
+) -> bool:
+    """Tells whether ``call`` has the same vararg ``*args`` as ``node``."""
+    vararg_name: Optional[str] = None
+    for ar in call.args:
+        # 'args': [<_ast.Starred object at 0x10d77a3c8>]
+        if isinstance(ar, ast.Starred):
```

Comment:
Can we reuse https://github.com/wemake-services/wemake-python-styleguide/blob/master/wemake_python_styleguide/logic/naming/name_nodes.py#L87 here?

Reply:
Seems not cause it will work differently for `ast.Name` in call.args.